components/dashboard/src/data/organizations/update-org-settings-mutation.ts

```diff
@@ -44,13 +46,18 @@ export const useUpdateOrgSettingsMutation = () => {
                 }
             }
 
-            const settings = await organizationClient.updateOrganizationSettings(update);
-            return settings.settings!;
+            const { settings } = await organizationClient.updateOrganizationSettings(update);
+            return settings!;
         },
-        onSuccess: () => {
-            invalidateOrgSettings();
+        onSuccess: (settings) => {
             invalidateWorkspaceClasses();
             invalidateOrgRepoSuggestions();
+
+            if (settings) {
+                queryClient.setQueryData(getQueryKey(organizationId), settings);
```

Comment:
this set's the query's data to not have to re-fetch data we have available to us. We do the same with updating Configurations